crates/ruff_dev/src/generate_ty_cli_reference.rs

```diff
@@ -80,6 +80,7 @@ fn generate() -> String {
 
     let mut parents = Vec::new();
 
+    output.push_str("<!-- WARNING: This file is auto-generated. Edit 'generate_ty_cli_reference.rs' instead. -->\n\n");
```

Comment:
It should only very rarely be necessary to edit the generation script itself. Most changes only need changes to `ty/args.rs`, followed by running `cargo dev generate-all`